### ISSUE TITLE
Possible bug on Regex?

### ISSUE THREAD

**Issue by RafaRuiz** [parse-community/parse-server#3965]
### Description of the problem

I'm not a regex expert, but when I've used it in the past, I usually used it for matching beginning / end / both of words or sentences. Example:
`.^nd.^` would match `nd`, `2nd`, `indoor`, `nda`.

In my case, I'm trying to match strict words independent of the lower/upper cases, and it doesn't seem to work.

I'm trying to match the user `maxjbv` in the server. Actually there are two users: `maxjbv` and `maxjbv2`. I want to log in the person independently if he types `maxjbv` or `mAxJBv`.

### Steps to reproduce
This is the actual code I use:
        
        var username = request.params.username || "";
	var password = request.params.password ||  "";

	var UsernameQuery = new Parse.Query("_User");
	UsernameQuery.matches("username", username, "i");
	var EmailQuery = new Parse.Query("_User");
	EmailQuery.matches("email", username, "i");

	var UserQuery = Parse.Query.or(UsernameQuery, EmailQuery);

And this is the `console.log(UserQuery);`:

`{"where":{"$or":[{"username":{"$regex":"maxjbv","$options":"i"}},{"email":{"$regex":"maxjbv","$options":"i"}}]}}`

As you see, there's no `.^` anywhere.

#### Expected Results
I'm expecting to fetch `maxjbv` from the server

#### Actual Outcome
In the query, `maxjbv2` is getting in the results

### Environment Setup

- **Server**
  - parse-server version (Be specific! Don't say 'latest'.) : 2.3.5
  - Operating System:     Ubuntu
  - Hardware:             DigitalOcean (1GB Ram machine) LON
  - Localhost or remote server? (AWS, Heroku, Azure, Digital Ocean, etc): DigitalOcean

- **Database**
  - MongoDB version: 3.4.1
  - Storage engine:  FSAdapter
  - Hardware:        DigitalOcean (1GB Ram machine) LON
  - Localhost or remote server? (AWS, mLab, ObjectRocket, Digital Ocean, etc): DigitalOcean

**Comment by flovilmart** [comment]
@RafaRuiz, could you write a failing test and open a PR with it?

**Comment by RafaRuiz** [comment]
@flovilmart any guide or post of how to do that a failing test here? I've never done it, I probably will do it in 2 secs but I need to compare it to something already done.

**Comment by flovilmart** [comment]
Into the spec folder, there should already be a bunch of tests around regex

**Comment by natanrolnik** [comment]
@RafaRuiz maybe try looking at these locations (not sure which one is where you need):

https://github.com/parse-community/parse-server/blob/master/spec/ParseQuery.spec.js#L1272

and

https://github.com/parse-community/parse-server/blob/df029b82ebf2776890d3e1fd1281fd0bf6ea9505/spec/QueryTools.spec.js#L306

**Comment by RafaRuiz** [comment]
Thank you for the support.

I've written the tests and they don't fail (then I'm not pushing).
So I suspected that it is because and old version of Parse Server (I had 2.3.5).
Actually, *2.5.0* doesn't work on my droplet, but *2.4.2* works ok, and this error doesn't happen.

Sorry to waste time, in the list of releases there wasn't anything related to what I was saying so I discarded the version problem.

Thank you!
